COMET/misc_plugins/PlotScripts/CONFIGS/importer/importer_PQC.py

```diff
@@ -0,0 +1,132 @@
+
+            #return a dictionary
+            return_dict[file_name] = {"data": {"Voltage": volt_dict[file_name], "Current": curr_dict[file_name]}}
+            return_dict[file_name]["header"] = [area[file_name][0], analysis_type]
+            return_dict[file_name].update({"measurements": ["Voltage", "Current"]})
+            return_dict[file_name].update({"units": ["V", "A"]})
+
+            number_of_files += 1
+
+        #importing a diode file, same as gate file just changing the file_name specification.
+        elif file_name[0:8] == 'CV_Diode':
+            capacity = []
+            volt = []
+            curr = []
+            for i in dic[file_name][start_data:(end_data + 1)]:
+                first, second, third = i.split('\t')
+                volt.append(first)
+                curr.append(second)
+                capacity.append(third)
+            volt_dict[file_name] = [float(i) for i in volt]
+            curr_dict[file_name] = [float(i) for i in curr]
+            capacity_dict[file_name] = [float(i) for i in capacity]
+
+            area[file_name] = [s for s in dic[file_name] if "Area" in s]
+            analysis_type = [s for s in dic[file_name] if "Run" in s][0].split(' ')[0]
+            return_dict[file_name] = {"data": {"Voltage": volt_dict[file_name], "Current": curr_dict[file_name], 'Capacity': capacity_dict[file_name]}}
+            return_dict[file_name]["header"] = [area[file_name][0], analysis_type]
+            return_dict[file_name].update({"measurements": ["Voltage", "Current", 'Capacity']})
+            return_dict[file_name].update({"units": ["V", "A", 'F']})
+
+            number_of_files += 1
+
+        #importing a mos file, same as importing the other kind of analysis, here else is used because so far just three analysis can be done.
+        else:
+            capacity = []
+            volt = []
+            curr = []
+            for i in dic[file_name][start_data:(end_data + 1)]:
+                first, second, third = i.split('\t')
+                volt.append(first)
+                curr.append(second)
+                capacity.append(third)
+            volt_dict[file_name] = [float(i) for i in volt]
+            curr_dict[file_name] = [float(i) for i in curr]
+            capacity_dict[file_name] = [float(i) for i in capacity]
+            area[file_name] = [s for s in dic[file_name] if "Area" in s]
+            analysis_type = [s for s in dic[file_name] if "Run" in s][0].split(' ')[0]
+            return_dict[file_name] = {"data": {"Voltage": volt_dict[file_name], "Current": curr_dict[file_name],
+                                               'Capacity': capacity_dict[file_name]}}
+            return_dict[file_name]["header"] = [area[file_name][0], analysis_type]
+            return_dict[file_name].update({"measurements": ["Voltage", "Current", 'Capacity']})
+            return_dict[file_name].update({"units": ["V", "A", 'F']})
+            number_of_files += 1
```

Comment:
These other cases of the if are doing basically the same as the first one. The only difference here is that the names and number of measurements. This screams for a function which yields you your data. This way the code gets clean, easier to read and change and is more structured.

the lines:

`first, second, third = i.split('\t')`
is not very pythonic

do:

`splitted = i.split()`

and access the party by splitted[0] etc.